sympy/solvers/ode/single.py

```diff
@@ -867,8 +867,11 @@ def _matches(self):
         df = f(x).diff(x)
         self.eqs = []
         eq = eq_orig.collect(f(x), func = cancel)
-        eq = fraction(factor(eq))[0]
-        factors = Mul.make_args(factor(eq))
+        try:
+            eq = fraction(factor(eq))[0]
+            factors = Mul.make_args(eq)
+        except RecursionError:
+            return False
```

Comment:
This should not catch `RecursionError`. What causes the RecursionError?

It is almost certainly a bug that should be fixed.